### PR TITLE
machine starting status

### LINKED ISSUE #13711: podman machine list report machine being running before machine is really usable
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Start a podman machine (like the default one) using `podman machine start`

And check 'podman machine list' result
You'll notice that the machine goes in state 'Running' before that the machine is really Running.

**Steps to reproduce the issue:**

1. run script `while true;do clear; podman machine list; sleep 1;done` in one terminal

2.`podman machine start` in another terminal

3. Check that the machine goes in 'up/running' state before we see the message stating the machine is running in the terminal where we launched the podman machine.

https://user-images.githubusercontent.com/436777/160791334-9a17621c-0b72-468f-8682-9027c31b9a6e.mp4

**Describe the results you received:**

Running state is displayed too early. For example socket isn't available while it says it's 'Running'

**Describe the results you expected:**

Machine should be in `Running` state when we see the message

```
Machine "podman-machine-default" started successfully
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
4.0.2 / macOS
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by cdoern**
podman machine was using the file modification time to get the running status
add three new config entries Starting (bool) Created (time) LastUp (time) to actually
keep track of when these events happened. This means we can use the config file
to actually store this data and not mess up the created/last-up time.

This fixes the issues where the machine would report running 15 seconds before it was up.
Also fixes the issue of modifying the file manually and saying the machine is "up"

[NO NEW TESTS NEEDED]

resolves #13711

Signed-off-by: cdoern <cbdoer23@g.holycross.edu>

### CODE REVIEW

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -630,18 +644,21 @@ func (v *MachineVM) Start(name string, _ machine.StartOptions) error {
 	}
 
 	v.waitAPIAndPrintInfo(forwardState, forwardSock)
-
+	//	v.Starting = false
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:647]: What is this?

File: pkg/machine/qemu/machine.go [hunk 2]
```diff
@@ -630,18 +644,21 @@ func (v *MachineVM) Start(name string, _ machine.StartOptions) error {
...
 	return nil
 }
 
 func (v *MachineVM) checkStatus(monitor *qmp.SocketMonitor) (machine.Status, error) {
 	// this is the format returned from the monitor
 	// {"return": {"status": "running", "singlestep": false, "running": true}}
 
+	// REMEMBER
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:655]: The Alamo?

Comment (cdoern) [reply pkg/machine/qemu/machine.go:655]: Agh thought I got all of my comments, will do another sweep

File: pkg/machine/qemu/machine.go (hunk 1 above)
Comment (cdoern) [reply pkg/machine/qemu/machine.go:647]: leftover as well thanks for catching that. I Opted for the `defer` instead.

**Comment by baude** [comment]
we now have e2e tests for machine in `pkg/machine/e2e`. I would be willing to help you with them or we could setup some pair programming time.

**Comment by rhatdan** [comment]
LGTM
Can we merge this and add tests in a separate PR.  Would like to get feedback from podman desktop team.

**Comment by baude** [comment]
@cdoern you owe me tests ...

**Comment by cdoern** [comment]
@baude I have some tests in the works for this! I will hopefully have them done today. Is it ok if I ping you with questions as I go along?

**Pull request merged**